src/services/exit_order.py

```diff
@@ -1,24 +1,334 @@
+            validator for validator in self.w3.cc.get_validators(blockstamp.state_root)
+            if not self._is_on_exit(validator)
+        ])
+        self.lido_node_operator_stats = self._prepare_lido_node_operator_stats(
+            operators, operator_validators
+        )
 
-        # Iterator should contain only next validators
-        self.max_size = max_size
+    def _no_index_by_validator(self, validator: LidoValidator) -> NodeOperatorIndex:
+        return StakingModuleId(self.staking_module_id[validator.key.moduleAddress]), NodeOperatorId(validator.key.operatorIndex)
 
     def __iter__(self):
         return self
 
     def __next__(self):
+        while self.left_queue_count < self.max_validators_to_exit:
+            self.exitable_lido_validators.sort(key=lambda validator: self._predicates(validator))
+            to_exit = self.exitable_lido_validators.pop(0)
+            self._decrease_node_operator_stats(to_exit)
+            self.left_queue_count += 1
+            return to_exit
         raise StopIteration
+
+    def _decrease_node_operator_stats(self, validator: LidoValidator) -> None:
+        """
+        Sub particular validator stats from its node operator stats
+        We do it every time when validator is popped from the queue for resort the rest of queue
+        """
+        module_operator = self._no_index_by_validator(validator)
+        self.total_predictable_validators_count -= 1
+        self.lido_node_operator_stats[module_operator].predictable_validators_count -= 1
+        self.lido_node_operator_stats[module_operator].predictable_validators_total_age -= max(
+            0, self.blockstamp.ref_epoch - int(validator.validator.validator.activation_epoch)
+        )
+
+    # -- Predicates for sorting validators --
+    def _predicates(self, validator: LidoValidator) -> tuple:
+        module_operator = self._no_index_by_validator(validator)
+        operator_stats = self.lido_node_operator_stats[module_operator]
+        return (
+            # positive mean asc sorting
+            # negative mean desc sorting
+            self._operator_delayed_validators(operator_stats),
+            -self._operator_targeted_validators_to_exit(operator_stats),
+            -self._operator_stake_weight(operator_stats, self.total_predictable_validators_count),
+            -self._operator_predictable_validators(operator_stats),
+            self._validator_activation_epoch(validator),
+            self._validator_index(validator),
+        )
+
+    @staticmethod
+    def _operator_delayed_validators(operator_state: NodeOperatorPredictableState) -> int:
+        return operator_state.delayed_validators
+
+    @staticmethod
+    def _operator_targeted_validators_to_exit(operator_state: NodeOperatorPredictableState) -> int:
+        if operator_state.targeted_validators is None:
+            return 0
+        return operator_state.predictable_validators_count - operator_state.targeted_validators
```

Comment:
```suggestion
        return max(0, operator_state.predictable_validators_count - operator_state.targeted_validators)
```

Reply:
If we have: 
operator1: 200 validators, limit 1000
operator2: 200 validators, no limit
operator3: 200 validators, limit 200
operator4: 200 validators, limit 0

In current implementation we'll get exit order: [operator4, operator3, operator2, operator1]
In your suggestion we'll get exit order: [operator4, operator3 or operator1, operator2]

Why we should kick operator's validator who have `limit > predictable validators` one of the firsts?

Reply:
Found out on call with protocol team. Will be corrected as you said. I’ll also fix the naming

Reply:
fixed